src/render/painter.js

```diff
@@ -849,23 +849,31 @@ class Painter {
         if (fog && fogOpacity !== 0.0) {
             const temporalOffset = (this.frameCounter / 1000.0) % 1;
             const fogColor = fog.properties.get('color');
+            const fogColorUnpremultiplied = fogColor.a === 0 ? [0, 0, 0] : [
```

Comment:
I chose to validate that `fog.color.a !== 0` rather than `fog.color.a === 1` since we can pretty easy un-premultiply and leave most opacities at worst just a no-op.